### PR TITLE
[Udemy] Fix issue 8753

### PR DESCRIPTION
**Pull request by quanhua92**
Fix issue in https://github.com/rg3/youtube-dl/issues/8753
The web page changes the HTML in new version of video player so it causes the error of "Unable to extract course id". I change the regex and everything works now.

### CODE REVIEW

File: youtube_dl/extractor/udemy.py [hunk 1]
```diff
@@ -142,9 +142,12 @@ def _real_extract(self, url):
         lecture_id = self._match_id(url)
 
         webpage = self._download_webpage(url, lecture_id)
-
-        course_id = self._search_regex(
+        try:
+            course_id = self._search_regex(
             r'data-course-id=["\'](\d+)', webpage, 'course id')
```
Comment (dstftw) [inline youtube_dl/extractor/udemy.py]: Pass tuple of regexes to `_search_regex` instead.

**Comment by Cmakota** [comment]
Thank you sir!

File: youtube_dl/extractor/udemy.py (hunk 1 above)
Comment (quanhua92) [inline youtube_dl/extractor/udemy.py]: Sorry. I am new to this project and don't know how to do that.

Comment (dstftw) [inline youtube_dl/extractor/udemy.py]:
```python
course_id = self._search_regex(
    (r'data-course-id=["\'](\d+)', r'&quot;id&quot;: (\d+)'),
    webpage, 'course id')
```

Comment (quanhua92) [inline youtube_dl/extractor/udemy.py]: Thanks. Just fix this.

**Comment by quanhua92** [comment]
@Cmakota Just fix it with tuble search_regex. It is working right now. Until this pull request is merged into a release, you can change the udemy.py as you said to use this fix.

**Pull request closed without merging**